soda/bigquery/soda/data_sources/bigquery_data_source.py

```diff
@@ -80,6 +81,10 @@ def __init__(self, logs: Logs, data_source_name: str, data_source_properties: di
         if account_info_json_str:
             self.account_info_dict = json.loads(account_info_json_str)
 
+        if self.account_info_dict is None:
+            self.logs.info("Using application default credentials.")
+            self.credentials, _ = default()
```

Comment:
`self.credentials` is set to None a few lines below so this is not really taken into account anyway. I would suggest removing this `if` and doing 
```
if self.data_source_properties.get("use_context_auth") or self.account_info_dict is None:
```
below

Reply:
Agree, this is redundant. Will adjust it accordingly